### PR TITLE
[Datastore] Fix data types diffs get stats table

### PR DESCRIPTION
**Pull request by benbd86**
changed logic to handle different data types according to schema received from big query response.
supporting the common types:
        "STRING": "object",
        "BOOL": "bool",
        "FLOAT": "float64",
        "INTEGER": pd.Int64Dtype(),
        "TIMESTAMP": "datetime64[ns]",

Added test dedicated to google's big query. Requires a credentials file added to tests/system.

### CODE REVIEW

File: tests/system/feature_store/test_google_big_query.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import mlrun.feature_store as fs
+from mlrun.datastore.sources import BigQuerySource
+from mlrun.datastore.targets import ParquetTarget
+from tests.system.base import TestMLRunSystem
+
+CREDENTIALS_ENV = "GOOGLE_BQ_CREDENTIALS_JSON"
+
+
+def _is_env_params_doesnt_exist() -> bool:
```
Comment (Hedingber) [inline tests/system/feature_store/test_google_big_query.py:14]:
```suggestion
def _are_google_credentials_set() -> bool:
```

File: tests/system/feature_store/test_google_big_query.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            "BigQuerySource",
+            table="bigquery-public-data.chicago_taxi_trips.taxi_trips",
+            max_results_for_table=max_results * 2,
+            materialization_dataset="chicago_taxi_trips",
+            chunksize=max_results,
+        )
+        self._big_query_source_test("table_c", source, max_results)
+
+    @staticmethod
+    def _big_query_source_test(name: str, source: BigQuerySource, max_results: int):
```
Comment (Hedingber) [inline tests/system/feature_store/test_google_big_query.py:73]:
```suggestion
    def _test_big_query_source(name: str, source: BigQuerySource, max_results: int):
```

File: tests/system/feature_store/test_google_big_query.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+# Marked as enterprise because of v3io mount and pipelines
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.skipif(
+    _is_env_params_doesnt_exist(),
+    reason=f"Environment variable {CREDENTIALS_ENV} is not defined, skipping...",
+)
+@pytest.mark.enterprise
+# NOTE: this test uses a file for credentials called 'google_application_credentials.txt',
+#       make sure the file exists under 'tests/system/' and that the test runs on it as working directory.
```
Comment (Hedingber) [inline tests/system/feature_store/test_google_big_query.py:26]: The test's code creates this file, I think this is redundant no ?

File: tests/system/feature_store/test_google_big_query.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+import pandas as pd
+import pytest
+
+import mlrun.feature_store as fs
+from mlrun.datastore.sources import BigQuerySource
+from mlrun.datastore.targets import ParquetTarget
+from tests.system.base import TestMLRunSystem
+
+CREDENTIALS_ENV = "GOOGLE_BQ_CREDENTIALS_JSON"
```
Comment (Hedingber) [inline tests/system/feature_store/test_google_big_query.py:11]:
```suggestion
CREDENTIALS_ENV = "MLRUN_SYSTEM_TESTS_GOOGLE_BIG_QUERY_CREDENTIALS_JSON"
```

**Review by Hedingber — CHANGES_REQUESTED** [review]

**Review by Hedingber — APPROVED** [review]

**Review by yaronha — APPROVED** [review]

File: automation/system_test/prepare.py [hunk 5]
```diff
@@ -433,6 +435,7 @@ def main():
 @click.argument("access-key", type=str, required=True)
 @click.argument("iguazio-version", type=str, default=None, required=True)
 @click.argument("spark-service", type=str, required=True)
+@click.argument("google_big_query_credentials_json", type=str, required=True)
```
Comment (Hedingber) [inline automation/system_test/prepare.py:438]: Required should be False, please fix in a followup PR

**Review by Hedingber — APPROVED** [review]

**Pull request merged**